daphne/src/testing.rs

```diff
@@ -467,54 +471,48 @@ impl DapLeader<BearerToken> for MockAggregator {
 
     async fn put_report(&self, report: &Report) -> Result<(), DapError> {
         let task_id = &report.task_id;
-        let bucket_info = BucketInfo::new(
-            self.tasks
-                .get(task_id)
-                .ok_or_else(|| DapError::fatal("task not found"))?,
-            task_id,
-            &report.nonce,
-        );
+        let task_config = self
+            .get_task_config_for(task_id)
+            .ok_or_else(|| DapError::fatal("task not found"))?;
+
+        // Check whether the report is collected via the AggStore.
```

Comment:
The code for batch-collected and report-replayed checks for Leader and Helper look pretty similar. Is there any way to refactor so that they share code?

Reply:
The newest commit has implemented a function so that they share code.